src/components/HorizontalScroller.js

```diff
@@ -0,0 +1,105 @@
+import React, { useState, useRef, useEffect } from 'react'
+import styles from './HorizontalScroller.module.scss'
+
+export const HorizontalScroller = ({
+  children,
+  showArrows,
+  className,
+  classes
+}) => {
+  const [isNextDisabled, setIsNextDisabled] = useState(false)
+  const [isPrevDisabled, setIsPrevDisabled] = useState(true)
+  const [showGalleryNav, setShowGalleryNav] = useState(true)
+  const galleryRef = useRef(null)
+  const slideRef = useRef(null)
+
+  const [windowWidth, setWindowWidth] = useState(
+    typeof window !== 'undefined' ? window.innerWidth : 0
+  )
+
+  const handleScroll = () => {
+    const gallery = galleryRef.current
+    gallery.scrollLeft > gallery.scrollWidth - gallery.clientWidth - 15
+      ? setIsNextDisabled(true)
+      : setIsNextDisabled(false)
+    gallery.scrollLeft < 20 ? setIsPrevDisabled(true) : setIsPrevDisabled(false)
+  }
+
+  const handlePrev = () => {
+    const gallery = galleryRef.current
+    const slide = slideRef.current
+    gallery.scrollBy({ left: -(slide.offsetWidth + 20) }) // move gallery width of slide plus the margin for safari
+  }
+
+  const handleNext = () => {
+    const gallery = galleryRef.current
+    const slide = slideRef.current
+    gallery.scrollBy({ left: slide.offsetWidth + 20 }) // move gallery width of slide plus the margin for safari
```

Comment:
I think we should dynamically get the left and right margin of the slide instead of hard-coding `20`.